support/util/util.go

```diff
@@ -245,3 +246,69 @@ func ConvertImageRegistryOverrideStringToMap(envVar string) map[string][]string
 
 	return imageRegistryOverrides
 }
+
+// IsIPv4 function parse the CIDR and get the IPNet struct if the IPNet.IP cannot be converted to 4bytes format,
+// the function returns nil, if it's an IPv6 it will return nil.
+func IsIPv4(cidr string) (bool, error) {
+	_, ipnet, err := net.ParseCIDR(cidr)
+	if err != nil {
+		return false, fmt.Errorf("error validating the ClusterNetworkCIDR from HostedCluster: %v", err)
+	}
+
+	if ipnet.IP.To4() != nil {
+		return true, nil
+	} else {
+		return false, nil
+	}
+}
+
+// GetIpsForSvcNetsWithDefaults return the first IP of the serviceNetwork associated to the hcp
+// it covers Single and Dual stack. If the ServiceNet is not parseable, the function returns
+// 172.20.0.1 in IPv4 case and fd02::1 for IPv6
+func GetIpsForSvcNetsWithDefaults(svcNets []hyperv1.ServiceNetworkEntry, defaults *DefaultAdvIps) []string {
+	addresses := make([]string, 0)
+	for _, svcNet := range svcNets {
+		firstAddress, err := GetFirstUsableIP(svcNet.CIDR.String())
+		if err != nil {
+			fmt.Println("error getting the first usable IP:", err)
+			continue
+		}
+		// Because of this issue https://github.com/kubernetes/enhancements/issues/2438 KAS does not work in dual stack
+		// so the AdvertiseAddress even being an slice, only supports 1 item, if more are included it will use the last one.
+		addresses = append(addresses, firstAddress)
+		// ToDo (jparrill): When the mentioned issue get solved, remove the break to add all the addresses in the slice.
+		break
+	}
+
+	// I cannot think in a scenario where the serviceNetworks are not rightly parseable
+	// but as a secure measure, if there is not any stored addresses at this point, we set
+	// IPv4 and IPv6 as a default ones.
+	if len(addresses) == 0 {
+		fmt.Println("error parsing defined ServiceNetworkEntries, defaulting AdvertiseAddress to IPv4:", defaults.IPv4)
+		addresses = append(addresses, defaults.IPv4)
+	}
+
+	finalList := RemoveDuplicatesFromList(addresses)
```

Comment:
your list can only have 1 item at most, why do you need  `RemoveDuplicatesFromList(addresses)` here?